src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,447 @@
+                                                                                                  drafts) {
+        final Set<String> skus = extractSkus(drafts);
+        return inventoryService.fetchInventoryEntriesBySkus(skus)
+            .thenApply(existingEntries -> existingEntries.stream().collect(toMap(SkuChannelKeyTuple::of,
+                entry -> entry)));
+    }
+
+    /**
+     * Returns distinct SKUs present in {@code inventories}.
+     *
+     * @param inventories {@link List} of {@link InventoryEntryDraft} where each draft contains its sku
+     * @return {@link Set} of distinct SKUs found in {@code inventories}.
+     */
+    private Set<String> extractSkus(final List<InventoryEntryDraft> inventories) {
+        return inventories.stream()
+                .map(InventoryEntryDraft::getSku)
+                .collect(Collectors.toSet());
+    }
+
+    /**
+     * Tries to update {@code entry} in CTP project with data from {@code draft}.
+     * It calculates list of {@link UpdateAction} and calls API only when there is a need.
+     * Before calculate differences, the channel reference from {@code draft} is replaced, so it points to
+     * proper channel ID in target system.
+     * It either updates inventory entry and increments "updated" counter in statistics, or increments "failed" counter
+     * and executes error callback function in case of any exception.
+     *
+     * @param entry entry from existing system that could be updated.
+     * @param draft draft containing data that could differ from data in {@code entry}.
+     *              <strong>Sku isn't compared</strong>
+     * @return {@link CompletionStage} of {@link Void} that indicates method progress
+     */
+    @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
+    private CompletionStage<Void> attemptUpdate(final InventoryEntry entry, final InventoryEntryDraft draft) {
+        final Optional<InventoryEntryDraft> fixedDraft = replaceChannelReference(draft);
+        if (fixedDraft.isPresent()) {
+            final List<UpdateAction<InventoryEntry>> updateActions =
+                InventorySyncUtils.buildActions(entry, fixedDraft.get(), syncOptions, typeService);
+            if (!updateActions.isEmpty()) {
+                return inventoryService.updateInventoryEntry(entry, updateActions)
+                    .thenAccept(updatedEntry -> statistics.incrementUpdated())
+                    .exceptionally(ex -> {
+                        statistics.incrementFailed();
+                        handleFailure(format("Failed to update inventory entry of sku '%s' and supply channel key '%s'",
+                            draft.getSku(), SkuChannelKeyTuple.of(draft).getKey()), ex);
+                        return null;
+                    });
+            }
+        } else {
+            statistics.incrementFailed();
```

Comment:
no, actually this covers only the attemptCreate case since this is a not existing inventory entry in your mocked list of inventory entries